### PR TITLE
[Refactor] Extract buildGithubText

### PR DESCRIPTION
**Pull request by smoralesd**
#### Description of changes

The markdown we get from `buildGithubText` is not for github only, we are going to use the same markdown for Azure Boards.
In this PR:

- Extract `buildGithubText` to it's own file
- Add unit tests for the new function
- Update `src\bug-filing\github\create-github-bug-filing-url.ts` (and the related tests) to use the new function

#### Pull request checklist

- [ ] Addresses an existing issue: Part of WI #1505514
- [x] Added relevant unit test for your changes. (`npm run test`)
- [x] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [x] Ran precheckin (`npm run precheckin`)
- [ ] Added screenshots/GIFs for UI changes.

### CODE REVIEW

File: src/bug-filing/common/get-issue-details-markdown.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { EnvironmentInfo } from '../../common/environment-info-provider';
+import { CreateIssueDetailsTextData } from '../../common/types/create-issue-details-text-data';
+import { IssueUrlCreationUtils } from './issue-filing-url-string-utils';
+
+export type IssueDetailsGetter = (
```
Comment (haonliu) [inline src/bug-filing/common/get-issue-details-markdown.ts]:
```suggestion
export type IssueDetailsMarkDownGetter = (
```

Comment (haonliu) [reply src/bug-filing/common/get-issue-details-markdown.ts]: Exploring 'suggested change' feature, I think we should use it more. :-)

Comment (smoralesd) [reply src/bug-filing/common/get-issue-details-markdown.ts]: I agree. One caveat thought: sometimes a rename suggestion may not rename all usages of the symbol being change, resulting in build/test failures.

File: src/bug-filing/common/get-issue-details-markdown.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    stringUtils: IssueUrlCreationUtils,
+    environmentInfo: EnvironmentInfo,
+    data: CreateIssueDetailsTextData,
+) => string;
+
+export const getIssueDetailsMarkdown = (
```
Comment (peterdur) [inline src/bug-filing/common/get-issue-details-markdown.ts]: Declare type explicitly to detect any future signature divergence?
```suggestion
export const getIssueDetailsMarkdown: IssueDetailsGetter = (
```

**Pull request merged**